backend/src/sysupport/custom.js

```
@@ -0,0 +1,22 @@
+import config from '../config'
+import rp from 'request-promise'
+
+export async function getCustomer(timedToken, user) {
+  const { host, prefix } = config.services.timed
+
+  const res = await rp({
+    method: 'get',
+    uri: `${host}${prefix}/customers`,
+    headers: {
+      accept: 'application/vnd.api+json',
+      'content-type': 'application/vnd.api+json',
+      Authorization: `Bearer ${timedToken}`
+    },
+    qs: {
+      reference: user.attributes.shortname
+    },
+    json: true
+  })
+
+  return res.data.shift()
```

Why `shift()`?

---

The endpoint "should" only return one customer but we can't be sure. So just take the first one we got..? Or would you rather throw an exception (log an error)?

---

Ah, ok! I'm ok with just returning the first result, but I'd use `[0]` instead of shift because shift has sideeffects and is evil 😉